plugins/multifactorauth/email/src/Extension/Email.php

```diff
@@ -0,0 +1,588 @@
+			function (MfaTable $record)
+			{
+				return $record->method == 'email';
+			}
+		);
+
+		if (count($emailRecords))
+		{
+			return;
+		}
+
+		// Add the email Method
+		try
+		{
+			/** @var MVCFactoryInterface $factory */
+			$factory = $this->app->bootComponent('com_users')->getMVCFactory();
+			/** @var MfaTable $record */
+			$record = $factory->createTable('Mfa', 'Administrator');
+			$record->reset();
+
+			$timeStep = min(max((int) $this->params->get('timestep', 120), 30), 900);
+			$totp     = new Totp($timeStep, self::CODE_LENGTH, self::SECRET_KEY_LENGTH);
+
+			$record->save(
+				[
+					'method'  => 'email',
+					'title'   => Text::_('PLG_MULTIFACTORAUTH_EMAIL_LBL_DISPLAYEDAS'),
+					'options' => [
+						'key' => ($totp)->generateSecret(),
+					],
+					'default' => 0,
+				]
+			);
+		}
+		catch (Exception $event)
+		{
+			// Fail gracefully
+		}
+	}
+
+	/**
+	 * Decodes the options from a record into an options object.
+	 *
+	 * @param   MfaTable  $record  The record to decode
+	 *
+	 * @return  array
+	 * @since   __DEPLOY_VERSION__
+	 */
+	private function decodeRecordOptions(MfaTable $record): array
+	{
+		$options = [
+			'key' => '',
+		];
+
+		if (!empty($record->options))
+		{
+			$recordOptions = $record->options;
+
+			$options = array_merge($options, $recordOptions);
+		}
+
+		return $options;
+	}
+
+	/**
+	 * Creates a new TOTP code based on secret key $key and sends it to the user via email.
+	 *
+	 * @param   string     $key   The TOTP secret key
+	 * @param   User|null  $user  The Joomla! user to use
+	 *
+	 * @return  void
+	 * @throws  Exception
+	 * @since   __DEPLOY_VERSION__
+	 */
+	private function sendCode(string $key, ?User $user = null)
+	{
+		static $alreadySent = false;
+
+		// Make sure we have a user
+		if (!is_object($user) || !($user instanceof User))
+		{
+			$user = $this->app->getIdentity()
+				?: Factory::getContainer()->get(UserFactoryInterface::class)->loadUserById(0);
+		}
+
+		if ($alreadySent)
+		{
+			return;
+		}
+
+		$alreadySent = true;
+
+		// Get the API objects
+		$timeStep = min(max((int) $this->params->get('timestep', 120), 30), 900);
+		$totp     = new Totp($timeStep, self::CODE_LENGTH, self::SECRET_KEY_LENGTH);
+
+		// Create the list of variable replacements
+		$code = $totp->getCode($key);
+
+		$replacements = [
+			'code'     => $code,
+			'sitename' => $this->app->get('sitename'),
+			'siteurl'  => Uri::base(),
+			'username' => $user->username,
+			'email'    => $user->email,
+			'fullname' => $user->name,
+		];
+
+		try
+		{
+			$jLanguage = $this->app->getLanguage();
+			$mailer = new MailTemplate('plg_multifactorauth_email.mail', $jLanguage->getTag());
+			$mailer->addRecipient($user->email);
+			$mailer->addTemplateData($replacements);
+			$mailer->send();
+		}
+		catch (MailDisabledException | phpMailerException $exception)
+		{
+			try
+			{
+				Log::add(Text::_($exception->getMessage()), Log::WARNING, 'jerror');
+			}
+			catch (\RuntimeException $exception)
```

Comment:
```suggestion
			catch (RuntimeException $exception)
```
`RuntimeException` is imported